substrate/frame/support/src/lib.rs

```diff
@@ -1513,7 +1514,8 @@ pub mod pallet_macros {
 	/// 	pub trait Config: frame_system::Config {
 	/// 		/// The overarching event type.
 	/// 		#[pallet::no_default_bounds] // Default with bounds is not supported for RuntimeEvent
-	/// 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+	/// 		#[allow(deprecated)]
+	///         type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
```

Comment:
You can remove all lines, but we can add a different associated type for showing `#[pallet::no_default_bounds]`